lib/matplotlib/tests/test_datetime.py

```diff
@@ -508,11 +508,29 @@ def test_stackplot(self):
         fig, ax = plt.subplots(layout='constrained')
         ax.stackplot(dates, stacked_nums)
 
-    @pytest.mark.xfail(reason="Test for stairs not written yet")
     @mpl.style.context("default")
     def test_stairs(self):
-        fig, ax = plt.subplots()
-        ax.stairs(...)
+        mpl.rcParams["date.converter"] = 'concise'
+
+        start_date = datetime.datetime(2023, 12, 1)
+        time_delta = datetime.timedelta(days=1)
+        baseline_date = datetime.datetime(1980, 1, 1)
+
+        bin_edges = [start_date + i * time_delta for i in range(31)]
+        edge_int = [0 + i * 1 for i in range(31)]
+        np.random.seed(123456)
+        values1 = np.random.randint(1, 100, 30)
+        random_day = np.random.randint(1, 10000, 30)
+        values2 = [start_date +
+                   datetime.timedelta(days=int(random_day[i])) for i in range(30)]
+        random_day = np.random.randint(-10000, 10000, 30)
+        values3 = [start_date +
+                   datetime.timedelta(days=int(random_day[i])) for i in range(30)]
```

Comment:
There's no need for `random_day` here:
```suggestion
        values2 = [start_date + datetime.timedelta(days=int(i))
                   for i in np.random.randint(1, 10000, 30)]
        values3 = [start_date + datetime.timedelta(days=int(i))
                   for i in np.random.randint(-10000, 10000, 30)]
```